src/solver/optimisation/adequacy_patch.h

```diff
@@ -0,0 +1,36 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#ifndef __SOLVER_ADEQUACY_FUNCTIONS_H__
+#define __SOLVER_ADEQUACY_FUNCTIONS_H__
+
+uint SetNTCForAdequacyFirstStep(bool AdequacyFirstStep,
+                                uint StartNodeAdequacyPatchType,
+                                uint EndNodeAdequacyPatchType,
+                                bool SetToZero12LinksForAdequacyPatch,
+                                bool SetToZero11LinksForAdequacyPatch);
```

Comment:
We're missing the documentation of the function, I propose:
```
/**
 * Determines if a link capacity needs to be set to 0. Only changes something if used during the AdequacyFirstStep.
 *
 * @param AdequacyFirstStep boolean for the first run of the optimization used by the adequacy patch
 * 
 * @param StartNodeAdequacyPatchType uint: The adq type of the node at the start of the link.
 * 
 * @param EndNodeAdequacyPatchType uint: The adq type of the node at the end of the link.
 * 
 * @param SetToZero12LinksForAdequacyPatch bool: Switch to cut links from nodes of adq type 1 towards nodes of adq type 2
 * 
 * @param SetToZero11LinksForAdequacyPatch bool: Switch to cut links from nodes of adq type 1 towards nodes of adq type 1
 * 
 * @return uint from an enumeration that describes the type of restrictions to put on this link for adq purposes.
 */
```

Reply:
Done.